beacon-chain/types/crystallized_state.go

```diff
@@ -265,15 +266,32 @@ func (c *CrystallizedState) NewStateRecalculations(aState *ActiveState, block *B
 		return nil, err
 	}
 
-	// TODO(471): Update rewards and penalties scheme to align with latest spec.
 	var rewardedValidators []*pb.ValidatorRecord
-	if len(block.Attestations()) != 0 {
-		rewardedValidators, _ = casper.CalculateRewards(
-			block.Attestations(),
+
+	// walk through all the slots from LastStateRecalc to LastStateRecalc + cycleLength - 1.
```

Comment:
Danny mentioned this is changed with the latest spec. Should be a cycleLength behind